File: VB/labelControlVB/Form1.vb
Imports DevExpress.XtraEditors

Public Class Form1

    Private Sub TextEdit1_EditValueChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles TextEdit1.EditValueChanged
        MyLabelControl1.Text = CType(sender, TextEdit).Text
    End Sub
End Class

File: CS/labelControl/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors;

namespace labelControl
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void textEdit1_EditValueChanged(object sender, EventArgs e)
        {
            myLabelControl1.Text = ((TextEdit)sender).Text;
        }
    }
}